tests/testthat/test-colsums_rowsums_linter.R

```diff
@@ -0,0 +1,96 @@
+test_that("colsums_rowsums_linter skips allowed usages", {
+  linter <- colsums_rowsums_linter()
+
+  expect_lint("apply(x, 1, prod)", NULL, linter)
+
+  expect_lint("apply(x, 1, function(i) sum(i[i > 0]))", NULL, linter)
+
+  # sum as FUN argument
+  expect_lint("apply(x, 1, f, sum)", NULL, linter)
+
+  # mean() with named arguments other than na.rm is skipped because they are not
+  # implemented in colMeans() or rowMeans()
+  expect_lint("apply(x, 1, mean, trim = 0.2)", NULL, linter)
+})
+
+
+test_that("colsums_rowsums_linter simple disallowed usages", {
+  linter <- colsums_rowsums_linter()
+  lint_message <- rex::rex("rowSums(x)")
+
+  expect_lint("apply(x, 1, sum)", lint_message, linter)
+
+  expect_lint("apply(x, MARGIN = 1, FUN = sum)", lint_message, linter)
+
+  expect_lint("apply(x, 1L, sum)", lint_message, linter)
+
+  expect_lint("apply(x, 1:4, sum)", rex::rex("rowSums(x, dims = 4)"), linter)
+
+  expect_lint("apply(x, 2, sum)", rex::rex("rowSums(colSums(x))"), linter)
+
+  expect_lint("apply(x, 2:4, sum)", rex::rex("rowSums(colSums(x), dims = 3)"), linter)
+
+  lint_message <- rex::rex("rowMeans")
+
+  expect_lint("apply(x, 1, mean)", lint_message, linter)
+
+  expect_lint("apply(x, MARGIN = 1, FUN = mean)", lint_message, linter)
+
+  # Works with extra args in mean()
+  expect_lint("apply(x, 1, mean, na.rm = TRUE)", lint_message, linter)
+
+  lint_message <- rex::rex("colMeans")
+
+  expect_lint("apply(x, 2, mean)", lint_message, linter)
+
+  expect_lint("apply(x, 2:4, mean)", lint_message, linter)
+
+})
+
+test_that("colsums_rowsums_linter works even when MARGIN can't be parsed", {
+
+  linter <- colsums_rowsums_linter()
+  lint_message <- rex::rex("l1")
+
+  expect_lint("apply(x, seq(2, 4), sum)", lint_message, linter)
+
+  expect_lint("apply(x, m, sum)", lint_message, linter)
```

Comment:
If `m` is a non-contiguous vector (e.g. `m <- c(1, 4)`), this would be a false positive, right?

Reply:
Yes, should I limit it to cases where MARGIN is numeric?

Reply:
Since you already have the `l1:l2` logic in your code, I'd refactor the XPath to allow NUM_CONSTANT and OP-COLON with two NUM_CONSTANT children.